algorithms/src/snark/marlin/marlin.rs

```diff
@@ -150,7 +150,7 @@ impl<E: PairingEngine, FS: AlgebraicSponge<E::Fq, 2>, MM: MarlinMode> MarlinSNAR
     ) -> FS {
         let mut sponge = FS::new_with_parameters(fs_parameters);
         sponge.absorb_bytes(&to_bytes_le![&Self::PROTOCOL_NAME].unwrap());
-        sponge.absorb_bytes(&batch_size.to_le_bytes());
+        sponge.absorb_bytes(&(batch_size as u64).to_le_bytes());
```

Comment:
@Pratyush is the batch size unsafe cast safe here?

Reply:
I don't see an issue here, it will not truncate

Reply:
Agreed, this is fine

Reply:
Though I would maybe use try_from to catch any edge cases.

Reply:
are any possible? I don't think so, unless we're thinking of some 128 bit architecture :thinking:?

Reply:
I think CHERI is one such architecture (though there are no production machines with CHERI enabled yet, I think)

Reply:
in that case it's fair to use `try_from` + `unwrap` as a futureproofing measure :+1:

Reply:
Basically what happens in the `as` case is that it wraps so it's basically `num%u64::MAX` (or whatever bit size you're working with).